tests/foreman/api/test_convert2rhel.py

```diff
@@ -0,0 +1,314 @@
+    job = entities.JobInvocation().run(
+        synchronous=False,
+        data={
+            'job_template_id': template_id,
+            'inputs': {
+                'Activation Key': activation_key_rhel.id,
+                'Restart': 'yes',
+            },
+            'targeting_type': 'static_query',
+            'search_query': f'name = {oracle.hostname}',
+        },
+    )
+    # wait for job to complete
+    wait_for_tasks(
+        f'resource_type = JobInvocation and resource_id = {job["id"]}', poll_timeout=1000
+    )
+    result = entities.JobInvocation(id=job['id']).read()
+    assert result.succeeded == 1
+
+    # check facts: correct os and valid subscription status
+    host_content = entities.Host(id=oracle.hostname).read_json()
+    assert (
+        host_content['operatingsystem_name'].startswith(f"RHEL Server {version}")
+        or host_content['operatingsystem_name'].startswith(f"RedHat {version}")
+        or host_content['operatingsystem_name'].startswith(f"RHEL {version}")
+    )
+    assert host_content['subscription_status'] == 0
+
+
+@pytest.mark.parametrize("version", ['centos7', 'centos8'], indirect=True)
+def test_convert2rhel_centos(centos, activation_key_rhel, version):
+    """Convert Centos linux to RHEL
+
+    :id: 6f698440-7d85-4deb-8dd9-363ea9003b92
+
+    :Steps:
+        0. Have host registered to Satellite
+        1. Check for operating system
+        2. Convert host to RHEL
+
+    :expectedresults: Host is converted to RHEL with correct os facts
+        and subscription status
+
+    :parametrized: yes
+
+    :CaseImportance: Medium
+    """
+    host_content = entities.Host(id=centos.hostname).read_json()
+    major = version.split('.')[0]
+    assert host_content['operatingsystem_name'] == f"CentOS {major}"
+
+    # execute job 'Convert 2 RHEL' on host
+    template_id = entities.JobTemplate().search(query={'search': 'name="Convert to RHEL"'})[0].id
+    job = entities.JobInvocation().run(
+        synchronous=False,
+        data={
+            'job_template_id': template_id,
+            'inputs': {
+                'Activation Key': activation_key_rhel.id,
+                'Restart': 'yes',
+            },
+            'targeting_type': 'static_query',
+            'search_query': f'name = {centos.hostname}',
+        },
+    )
+    # wait for job to complete
+    wait_for_tasks(
+        f'resource_type = JobInvocation and resource_id = {job["id"]}', poll_timeout=1000
+    )
+    result = entities.JobInvocation(id=job['id']).read()
+    assert result.succeeded == 1
+
+    # check facts: correct os and valid subscription status
+    host_content = entities.Host(id=centos.hostname).read_json()
+    assert (
+        host_content['operatingsystem_name'].startswith(f"RHEL Server {version}")
+        or host_content['operatingsystem_name'].startswith(f"RedHat {version}")
+        or host_content['operatingsystem_name'].startswith(f"RHEL {version}")
```

Comment:
Is this related to the [BZ2080347](https://bugzilla.redhat.com/show_bug.cgi?id=2080347)?
If so, please put the BZ link to the comment for reference.